presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -281,15 +281,51 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
     @Override
     public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
-        if (isPartitionsSystemTable(tableName)) {
-            return getPartitionsSystemTable(session, tableName);
+        if (partitionsTableHandler.matches(tableName)) {
+            return getPartitionsSystemTable(session, tableName, partitionsTableHandler.getSourceTableName(tableName));
+        }
+        else if (propertiesTableHandler.matches(tableName)) {
+            return getPropertiesSystemTable(session, tableName, propertiesTableHandler.getSourceTableName(tableName));
         }
         return Optional.empty();
     }
 
-    private Optional<SystemTable> getPartitionsSystemTable(ConnectorSession session, SchemaTableName tableName)
+    private Optional<SystemTable> getPropertiesSystemTable(ConnectorSession connectorSession, SchemaTableName tableName, SchemaTableName sourceTableName)
+    {
+        Optional<Table> table = metastore.getTable(sourceTableName.getSchemaName(), sourceTableName.getTableName());
+        if (!table.isPresent() || table.get().getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+            throw new TableNotFoundException(tableName);
+        }
+        Map<String, String> properties = table.get().getParameters();
+        List<ColumnMetadata> columns = properties.keySet().stream().map(key -> new ColumnMetadata(key, VarcharType.VARCHAR)).collect(toImmutableList());
```

Comment:
Let's sort the column names so they are returned in a consistent order:
```java
List<ColumnMetadata> columns = properties.keySet().stream()
        .sorted()
        .map(key -> new ColumnMetadata(key, VarcharType.VARCHAR))
        .collect(toImmutableList());

Reply:
done